lib/matplotlib/dates.py

```diff
@@ -1241,10 +1241,7 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
-        if interval < 1:
-            raise ValueError("The interval parameter must be an integer "
-                             "greater than or equal to one.")
-        if not isinstance(interval, int):
+        if interval < 1 or not isinstance(interval, int):
             raise ValueError("The interval parameter must be an integer "
                              "greater than or equal to one.")
```

Comment:
I think this can be condensed to: 
> interval must be an integer greater than 0

And totally absurdly nitpicky, but  I'd switch the error checks such that isinstance() check comes first so that the error checks match the order of the error message (which then in turn becomes the documentation of the error check)